BackEnd/src/main/java/com/unosquare/admin_core/back_end/repository/DashboardRepository.java

```diff
@@ -10,11 +10,19 @@
 
 
 public interface DashboardRepository extends JpaRepository<Event, Integer> {
-        @Query(
-        value = "SELECT * FROM Event e WHERE (e.start_Date >= :startDate AND e.end_Date <= :endDate) AND e.employee_id = :employeeId",
-        nativeQuery=true
-        )
+    @Query(
+            value = "SELECT * FROM Event e WHERE (e.start_Date >= :startDate AND e.end_Date <= :endDate) AND e.employee_id = :employeeId",
+            nativeQuery = true
+    )
     List<Event> findCalendarMonthEventsForEmployee(@Param("employeeId") int employeeId,
                                                    @Param("startDate") LocalDate startDate,
-                                                   @Param("endDate") LocalDate endDate );
+                                                   @Param("endDate") LocalDate endDate);
+
+    @Query(value = "SELECT * FROM Event e INNER JOIN Contract c  on e.employee_id = c.employee_id where (e.start_date >= :startDate " +
+                   "AND e.end_Date <= :endDate) AND c.team_id IN(select team_id FROM Contract c WHERE (c.employee_id = :employeeId " +
```

Comment:
Do we need to add checks on the top level contract for making sure the team members are still on the contract?

Reply:
SELECT * FROM Event e INNER JOIN Contract c on e.employee_id = c.employee_id 
where 
(
  (e.start_date BETWEEN :startDate AND :endDate) OR 
  (e.end_date BETWEEN :startDate AND :endDate) OR
  (e.start_date > :startDate AND e.end_date < :endDate)
)
AND 
(
  (c.start_date BETWEEN :startDate AND :today) OR 
  (c.end_date BETWEEN :today AND :endDate) OR
  (c.start_date < :startDate AND (:endDate IS NULL OR :endDate > :today)) 
)  
AND c.team_id IN 
(
  select team_id FROM Contract c 
  WHERE (c.employee_id = :employeeId AND
  (c.start_date BETWEEN :startDate AND :today) OR 
  (c.end_date BETWEEN :today AND :endDate) OR
  (c.start_date < :startDate AND (c.end_date IS NULL OR :endDate > :today))) 
)

Reply:
I suppose it would be beneficial down the line as the number of contracts/team increases, I will run through a couple of tests for this new query before I do a push.